### ISSUE TITLE
Version is not resolved correctly when alpha tag and version tag point to the same commit

### ISSUE THREAD

**Issue by Endron** [allegro/axion-release-plugin#149]
I run into a bug the other day. While preparing to release a new version of our application I added alpha tags to SCM to set the versions I wanted. I did this as shown in the documentation by executing 

```
./gradlew markNextVersion -Prelease.version=<next.version>
```

This created the tag as expected.
I then created a new release by running

```
./gradlew release -Prelease.disableUncommittedCheck -Prelease.disableRemoteCheck
```

This also tagged the version as expected.
Now we have an alpha tag and an release tag pointing to the same commit in the repository,

But then it went downhill when I tried to have the plugin resolve the version 

```
./gradlew currentVersion
```

Returned a SNAPSHOT version. As the version was what I initially set using "markNextVersion" it looks like the release tag was ignored. The problem with this is that now instead of building the release version I want to deploy to our repository manager (Artifactory) I am still building a SNAPSHOT version.

**Environment**
- Plugin version used was 1.3.4
- "markNextVersion" was executed on Windows 7 running with Java 8 and Gradle 2.12
- "release" was executed on a Linux CI server running Bamboo. Also Java 8 and Gradle 2.12
- the prefix for our release tags was "usl-nominator" and "batch-uploader" (I had this problem with two different applications in different git repos.)

**Workaround**
This does not happen when the alpha and the release tag do not point to the same git commit. So if you put some dummy commit between "markNextVersion" and "release" you are save.

**TL;DR**
If an alpha tag and an release tag point to the same git commit the version is not recognized correctly. The version is returned as SNAPSHOT when it should be the release version.

**Comment by adamdubiel** [comment]
Hi, this is the same issue as discussed in comments in #141 - i will be working on solution to this.

**Comment by adamdubiel** [comment]
Closed in 1.5.0 by #156

**Comment by radimk** [comment]
Note that #156 didn't fix it. Fortunatelly it is fixed not by https://github.com/allegro/axion-release-plugin/commit/634ff11e1c414d2df03e3cfb0587618b78e527e5 in 1.7.2 (which BTW is only available from mavenCentral and not from plugins.gradle.org repo as described in #197).
